### ISSUE TITLE
ECS - ServiceConnect : There is no way to reference ServiceConnect Discovery URL which is needed to build APIGatewayv2 Integration

### ISSUE THREAD

**Issue by anassar-lab** [aws/aws-cdk#31215]
### Describe the feature

Extend the CDK ability to build API Gateway Integration that references Service Connect, similar to the capability offered for Service Discovery, see the link: https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_apigatewayv2_integrations-readme.html#cloud-map-service-discovery.

This is currently possible from the console, using the Private integration target that references the ServiceConnect CloudMap entry:

![Screenshot 2024-08-26 114553](https://github.com/user-attachments/assets/1ebe76e3-ccdd-47f6-b227-a4ccaf0259bd)

Which generates an integration with the below properties:
<img width="469" alt="image (16)" src="https://github.com/user-attachments/assets/aac5a878-5161-4622-82a2-4af338563006">

### Use Case

Use CDK natively to build an integration between an ECS service that has service connect enabled and an API Gateway using private integration.

### Proposed Solution

There are a number of options, but I was trying to get a reference for the ServiceConnect Discovery URI and pass that as a parameter the HttpIntegration method, but I wasn't able to do that.
The ServiceConnect discovery URL can be viewed as shown below:
![Screenshot 2024-08-26 120330](https://github.com/user-attachments/assets/d999f1a7-b080-4037-b2b4-07f45875a009)

The URL can be constructed using  the pattern `serviceArn:'arn:aws:servicediscovery:<region-id:<aws_account_id>:service/<service_id>` which requires a reference to the CloudMap service ID.

```
const httpIntegration = new apigatewayv2.HttpIntegration(this, 'MyHttpIntegration', {
            httpApi: httpEndpoint,
            integrationType: apigatewayv2.HttpIntegrationType.HTTP_PROXY,
            connectionId: 'connectionId',
            connectionType: apigatewayv2.HttpConnectionType.VPC_LINK,
            integrationUri:  **___<This URI is needed>___**
            method: apigatewayv2.HttpMethod.ANY, 
            timeout: cdk.Duration.minutes(30), 
          });
```

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.148.0 (build e5740c0)

### Environment details (OS name and version, etc.)

Microsoft Windows 11 Enterprise - 23H2

**Comment by ashishdhingra** [comment]
Selecting service connect namespace is an option while adding API Gateway integration in AWS console. (Followed example https://github.com/aws-samples/ecs-service-connect-yelb-sample-app)

**Comment by zoro273** [comment]
@ashishdhingra , yes it is possible via console, see the screenshot attached in the feature description, but the main goal is to be able to accomplish the same via CDK.

**Comment by MirandaDora** [comment]
If not using CDK to provision ECS cluster, then it's possible to retrieve service using **fromServiceAttributes**:
```
const { Stack } = require('aws-cdk-lib');
// const sqs = require('aws-cdk-lib/aws-sqs');
const servicediscovery = require('aws-cdk-lib/aws-servicediscovery');
const apigatewayv2 = require('aws-cdk-lib/aws-apigatewayv2');
const ec2 = require('aws-cdk-lib/aws-ec2');
const { HttpServiceDiscoveryIntegration } = require('aws-cdk-lib/aws-apigatewayv2-integrations');;

class SD extends Stack {
  /**
   *
   * @param {Construct} scope
   * @param {string} id
   * @param {StackProps=} props
   */
  constructor(scope, id, props) {
    super(scope, id, props);
    const vpc = ec2.Vpc.fromLookup(this, 'MyVpc', {
      vpcId: 'vpcid'
    });
    const vpcLink = apigatewayv2.VpcLink.fromVpcLinkAttributes(this, 'servicediscoveryLink', {
      vpcLinkId: 'vpclinkid',
      vpc
    })
    const service = servicediscovery.Service.fromServiceAttributes(this, 'servicediscovery', {
      namespace: 'local',
      dnsRecordType: servicediscovery.DnsRecordType.A_AAAA,
      routingPolicy: servicediscovery.RoutingPolicy.WEIGHTED,
      serviceArn:'arn:aws:servicediscovery:ap-southeast-2:<aws_account_id>:service/<service_id>',
      serviceName: 'Demo',
      serviceId: 'service_id',
      discoveryType: servicediscovery.DiscoveryType.DNS_AND_API,
    })
    const httpEndpoint = new apigatewayv2.HttpApi(this, 'HttpProxyPrivateApi', {
      defaultIntegration: new HttpServiceDiscoveryIntegration('Labgroups', service, {
        vpcLink,
        method: apigatewayv2.HttpMethod.ANY
      }),
    });
  }
}
```
Also is that's possible to put some examples in the CDK documentation that does this?
Thank you.

**Comment by anassar-lab** [comment]
@MirandaDora _serviceId: 'service_id'_, attribute is also not retrievable from cloudformation or CDK, so this has to be passed in manually to the template after the service is created in a prior step, is my understanding correct ?
